src/CMakeLists.txt

```
@@ -0,0 +1,79 @@
+# set minimum cmake version
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+# Uncomment for VERBOSE Makefiles. Useful for debugging.
+#set(CMAKE_VERBOSE_MAKEFILE ON)
+
+# project name and language
+project(ImpactT LANGUAGES Fortran)
+
+option(USE_MPI "Activate MPI build" OFF)
+
+
+list(APPEND _sources
+    DataStruct/NumConst.f90
+    DataStruct/PhysConst.f90
+    DataStruct/Pgrid.f90
+	DataStruct/Data.f90
+	Func/Timer.f90
+	Func/Transpose.f90
+	Func/Fldmger.f90
+	Func/Ptclmger.f90
+	Func/FFT.f90
+    Appl/BPM.f90
+    Appl/CCL.f90
+    Appl/CCDTL.f90
+    Appl/DTL.f90
+    Appl/SC.f90
+    Appl/Multipole.f90
+	Appl/DriftTube.f90
+	Appl/Quadrupole.f90
+	Appl/ConstFoc.f90
+	Appl/SolRF.f90
+	Appl/Sol.f90
+	Appl/Dipole.f90
+	Appl/EMfld.f90
+	Appl/EMfldCart.f90
+	Appl/EMfldCyl.f90
+	Appl/EMfldAna.f90
+	Appl/BeamLineElem.f90
+	Appl/CompDom.f90
+	Appl/BeamBunch.f90
+	Appl/Depositor.f90
+	Appl/Ranger.f90
+	Appl/Field.f90
+	Appl/Distribution.f90
+	Contrl/Input.f90
+	Contrl/Output.f90
+	Contrl/AccSimulator.f90
+	Contrl/main.f90
+)
+
+if(USE_MPI)
+    set(OUTPUT_NAME "ImpactTexe-mpi")
+    add_executable(${OUTPUT_NAME})
+
+    find_package(MPI REQUIRED)
+    target_compile_definitions(${OUTPUT_NAME} PUBLIC USE_MPI)
+    target_link_libraries(${OUTPUT_NAME} PUBLIC MPI::MPI_Fortran)
+    include_directories(${MPI_Fortran_INCLUDE_PATH})
+else()
+    set(OUTPUT_NAME "ImpactTexe")
+    add_executable(${OUTPUT_NAME})
+
+    # Add mpistub to source list
+    list(APPEND _sources mpistub.f90)
+
+    # Create mpif.h from mpif_single.h
+    configure_file(mpif_single.h ${ImpactT_BINARY_DIR}/include/mpif.h COPYONLY)
+    # Add include dir above into search path
+    target_include_directories(${OUTPUT_NAME} PRIVATE $<BUILD_INTERFACE:${ImpactT_BINARY_DIR}/include>)
+endif()
```

I would simplify/splot like this:
```suggestion
if(USE_MPI)
    set(OUTPUT_NAME "ImpactTexe-mpi")
else()
    set(OUTPUT_NAME "ImpactTexe")
endif()
add_executable(${OUTPUT_NAME})

if(USE_MPI)
    find_package(MPI REQUIRED)
    target_compile_definitions(${OUTPUT_NAME} PUBLIC USE_MPI)
    target_link_libraries(${OUTPUT_NAME} PUBLIC MPI::MPI_Fortran)
    include_directories(${MPI_Fortran_INCLUDE_PATH})
else()
    # Add mpistub to source list
    list(APPEND _sources mpistub.f90)

    # Create mpif.h from mpif_single.h
    configure_file(mpif_single.h ${ImpactT_BINARY_DIR}/include/mpif.h COPYONLY)
    # Add include dir above into search path
    target_include_directories(${OUTPUT_NAME} PRIVATE $<BUILD_INTERFACE:${ImpactT_BINARY_DIR}/include>)
endif()
```

---

I had it like that before but I changed to a single use of the `if(USE_MPI)`. It gets much better for readability this way.